app/forms/mappings/Formatters.scala

```diff
@@ -131,8 +131,8 @@ trait Formatters {
       val maxLengthEmailAddress = 85
 
       (emailAddress, useEmail) match {
-        case (None, Some("01")) => Left(Seq(FormError(emailFieldName, keyRequired)))
         case (_, None) => Left(Seq(FormError(selectionFieldName, keySelectionRequired)))
+        case (Some(email), Some("01")) if email.length == 0 => Left(Seq(FormError(emailFieldName, keyRequired)))
```

Comment:
Hi @somaraja not sure why this has been changed. If the user selects Yes but doesn't enter an email the value is None. Can we go over this tomorrow together?